packages/dd-trace/src/plugins/database.js

```diff
@@ -1,9 +1,36 @@
 'use strict'
 
 const StoragePlugin = require('./storage')
+const TextMapPropagator = require('../opentracing/propagation/text_map.js')
+const Config = require('../../src/config')
+const config = new Config()
+const textMap = new TextMapPropagator(config)
+const carrier = {}
 
 class DatabasePlugin extends StoragePlugin {
   static get operation () { return 'query' }
+
+  createDBMPropagationCommentService () {
+    const dddbs = encodeURIComponent(this.config.service)
+    const dde = encodeURIComponent(this.tracer._env)
+    const ddps = encodeURIComponent(this.tracer._service)
+    const ddpv = encodeURIComponent(this.tracer._version)
+
+    return `dddbs='${dddbs}',dde='${dde}',ddps='${ddps}',ddpv='${ddpv}'`
+  }
+
+  injectDBMquery (query) {
+    config.experimental.traceparent = true
+    const serviceTags = this.createDBMPropagationCommentService()
+    textMap._injectTraceparent(this.activeSpan._spanContext, carrier)
```

Comment:
It can be unsafe to use a file variable to store information, especially if it will be injected in customer data. It would also be better in any case to extract the logic from `_injectTraceparent` to a separate helper that would be used by both `TextMapPropagator` and DBM instead of using a private method with an unnecessary carrier.

Reply:
@rochdev what do you mean by extract logic? Rebuild the logic in a helper or use the function within a helper to pull out what I need?

Reply:
@rochdev can we throw a static method on the `TextMapPropagator` class and call it a helper? Or do you think it would better belong in another location?

```js
static getTraceparent(spanContext) {
  const sampling = spanContext._sampling.priority >= AUTO_KEEP ? '01' : '00'
  const traceId = spanContext._traceId.toString(16).padStart(32, '0')
  const spanId = spanContext._spanId.toString(16).padStart(16, '0')
  return `01-${traceId}-${spanId}-${sampling}`
}
```

Reply:
@crysmags I meant to move the logic to a helper than is then imported and used by both `TextMapPropagator` and the plugin class.

@tlhunter I think it should probably go in `datadog-core` if it will be used in both plugin and dd-trace, especially with IDs and W3C tracecontext being core concepts for an APM, but I don't have a strong opinion since we can always move it later, but it shouldn't require exposing an internal class of the tracer.

Reply:
The important part is not really where it is, but that it should be private and it shouldn't require an object to be passed since that's unique to `TextMapPropagator`.

Reply:
Ok I created a helper in the db plugin in order to not use object and wrote a to do to create an additional helper that could be used by both text map and the plugins